spring/src/main/java/io/cloudevents/spring/messaging/CloudEventHeaderUtils.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents.spring.messaging;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.CloudEventContext;
+import io.cloudevents.SpecVersion;
+import io.cloudevents.core.CloudEventUtils;
+import io.cloudevents.core.builder.CloudEventBuilder;
+
+import org.springframework.messaging.MessageHeaders;
+
+import static io.cloudevents.spring.messaging.CloudEventsHeaders.CE_PREFIX;
+
+/**
+ * Utility class for copying message headers to and from {@link CloudEventContext}.
+ * 
+ * @author Dave Syer
+ *
+ */
+public class CloudEventHeaderUtils {
+
+	/**
+	 * Helper method for converting {@link MessageHeaders} to a {@link CloudEvent}. The
+	 * input headers must represent a valid event in "binary" form, i.e. it must have
+	 * headers "ce-id", "ce-specversion" etc.
+	 * @param headers the input request headers
+	 * @return a {@link CloudEventBuilder} that can be used to create a new
+	 * {@link CloudEvent}
+	 * 
+	 */
+	public static CloudEventBuilder fromMap(Map<String, ?> headers) {
```

Comment:
I don't really expect a builder as return value, i expect a `CloudEventContext`

Reply:
This usually would be called by the framework  (e.g., one of Spring frameworks) and it may want to add additional headers, attributes based on configuration, so returning an open builder is a pretty common pattern

Reply:
It's still weird, I've asked for a conversion from A to B, while here it's doing the conversion from A to B and then to a builder, which will return C. It's not symmetric with `toMap`, you can go from B to C easily and also IMO goes a little bit against the [single responsability principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).

Reply:
I think this is a remnant of the code before `CloudEventContext` existed. I can change it, I guess.